Source/NSFontCollection.m

```diff
@@ -0,0 +1,637 @@
+  NSArray *a = [self matchingDescriptorsWithOptions: options];
+  NSEnumerator *en = [a objectEnumerator];
+  id o = nil;
+
+  while ((o = [en nextObject]) != nil)
+    {
+      if ([[o familyName] isEqualToString: family])
+        {
+          [r addObject: o];
+        }    
+    }
+  
+  return [r copy];
+}
+
+- (instancetype) copyWithZone: (NSZone *)zone
+{
+  NSFontCollection *fc = [[NSFontCollection allocWithZone: zone] init];
+  [fc _setFontCollectionDictionary: _fontCollectionDictionary];
+  return fc;
+}
+
+- (instancetype) mutableCopyWithZone: (NSZone *)zone
+{
+  NSMutableFontCollection *fc = [[NSMutableFontCollection allocWithZone: zone] init];
+
+  [fc _setFontCollectionDictionary: [_fontCollectionDictionary mutableCopyWithZone: zone]];
+
+  return fc;
+}
+
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+  if ([coder allowsKeyedCoding])
+    {
+      [coder encodeObject: _fontCollectionDictionary
+                   forKey: @"NSFontCollectionDictionary"];
+    }
+  else
+    {
+      [coder encodeObject: _fontCollectionDictionary];
+    }
+}
+
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  self = [super init];
+  if (self != nil)
+    {
+      if ([coder allowsKeyedCoding])
+        {
+          ASSIGN(_fontCollectionDictionary,
+                 [coder decodeObjectForKey: @"NSFontCollectionDictionary"]);
+        }
+    }
+  else
+    {
+      [coder decodeValueOfObjCType: @encode(id) at: &_fontCollectionDictionary];
+    }
+  return self;
+}
+  
+@end
+ 
+
+@implementation NSMutableFontCollection
+
++ (NSMutableFontCollection *) fontCollectionWithDescriptors: (NSArray *)queryDescriptors
+{
+  return [[NSFontCollection fontCollectionWithDescriptors: queryDescriptors] mutableCopy];
+}
+
++ (NSMutableFontCollection *) fontCollectionWithAllAvailableDescriptors
+{
+  return [[NSFontCollection fontCollectionWithAllAvailableDescriptors] mutableCopy];
+} 
+
++ (NSMutableFontCollection *) fontCollectionWithLocale: (NSLocale *)locale
+{
+  return [[NSFontCollection fontCollectionWithLocale: locale] mutableCopy];
+}
+
++ (NSMutableFontCollection *) fontCollectionWithName: (NSFontCollectionName)name
+{
+  return [[NSFontCollection fontCollectionWithName: name] mutableCopy];
+}
+
++ (NSMutableFontCollection *) fontCollectionWithName: (NSFontCollectionName)name
+                                          visibility: (NSFontCollectionVisibility)visibility
+{
+  return [[NSFontCollection fontCollectionWithName: name visibility: visibility] mutableCopy];
+}
+
+- (NSArray *) queryDescriptors
+{
+  return [super queryDescriptors];
+}
+
+- (void) setQueryDescriptors: (NSArray *)queryDescriptors
+{
+  [super _setQueryDescriptors: [queryDescriptors mutableCopy]];
+}
+
+- (NSArray *) exclusionDescriptors
+{
+  return [_fontCollectionDictionary objectForKey: @"NSFontExclusionDescriptorAttributes"];
+}
+
+- (void) setExclusionDescriptors: (NSArray *)exclusionDescriptors
+{
+  NSMutableArray *qd = [[self queryDescriptors] mutableCopy];
```

Comment:
I just run a few tests on Cocoa and it looks like this method just sets the exclusion descriptors without looking at the query descriptors. You could drop the first three lines here.